orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -153,20 +165,23 @@ def load_features(self):
         self.display_indices = []
         if self.corpus is not None:
             domain = self.corpus.domain
+            self.closeContext()
```

Comment:
IMO closing and opening a context fits better to `set_data` than here. Also, context should be closed immediately after the data is removed from the widget, and not only when the next data set arrives. For example, load some data, pass it to CorpusViewer, set something and remove the data. Now restart Orange without saving the schema and observe that the selection is lost.